### PR TITLE
add "Use `.first()` for single element" rule

### LINKED ISSUE #16: "Use `.first()` for single element" guideline?
jQuery always returns a collection when using `$(selector)` or `$root.find(selector)`, while sometimes you are only interested in / only expect one element. In vanilla JS you would use `.querySelector(selector)` instead of `.querySelectorAll(selector)`. To clarify to other developers you only expect one element, I would suggest we always append `.first()`:

``` javascript
// collection of buttons:
$buttons = $form.find('button');
// versus just a single button:
$submitButton = $form.find('[type="submit"]').first();
```

Naturally variable names should also reflect this. So plural for collections (`$buttons`), singular for a individual element (`$button`).

Make this a guideline?

### PR DESCRIPTION
**Pull request by jpsc**
Closes #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -219,5 +220,22 @@ function toggleHidden(element) {
 	}
 }
 ```
+[↑ back to Table of Contents](#table-of-contents)
+
+## Use `.first()` for single element
+
+### Why?
```
Comment (jbmoelker) [inline README.md:344]: split `### Why?` text in summary / background (before Why) and reasons (after Why)? Reasons being: communicate to other devs you expect a single element. More reasons?

File: README.md [hunk 2]
```diff
@@ -219,5 +220,22 @@ function toggleHidden(element) {
...
+jQuery always returns a collection when using `$(selector)` or `$root.find(selector)`, while sometimes you are only interested in / only expect one element. In vanilla JS you would use `.querySelector(selector)` instead of `.querySelectorAll(selector)`. For clarity you should use [`.first()`](http://api.jquery.com/first/).
+
+### How?
```
Comment (jbmoelker) [inline README.md:347]: Describe "How?" in text before code snippet?: "Use `first()` to get the first item of the collection. Also ensure variable name reflects its a single element."

File: README.md [hunk 3]
```diff
@@ -219,5 +220,22 @@ function toggleHidden(element) {
...
+
+```javascript
+// collection of buttons:
+$buttons = $form.find('button');
```
Comment (jbmoelker) [inline README.md:351]: Do `$buttons.get()` and `$submitButton.get()` both return an array? If so, does that make this rule confusing?

Comment (jpsc) [inline README.md:351]: .first() returns a jQuery object. Whether you use .get() or nor later I think does not make **this** rule it confusing.

**Comment by jbmoelker** [comment]
@jpsc Since this is one of the selector rules I would put it before event handling, and after assign $.

**Pull request merged**